From: Dara Ramesh <dramesh@nvidia.com>
Date: Mon, 30 Apr 2018 09:05:17 +0000 (+0530)
Subject: extcon: Add new extcon types for display states
X-Git-Tag: tegra-l4t-r32.1~491
X-Git-Url: http://nv-tegra.nvidia.com/gitweb/?p=linux-4.9.git;a=commitdiff_plain;h=580d04cbdd48cd2840b91354f6b92a93b96e3350

extcon: Add new extcon types for display states

add new extcon types:

EXTCON_DISP_AUDIO_AUX0
EXTCON_DISP_AUDIO_AUX1
EXTCON_DISP_AUDIO_AUX2
EXTCON_DISP_AUDIO_AUX3

Here we are adding the AUDIO cable for each SOR
instance

Also adding the header file for the display
extcon driver.

TDS-3299
Bug 200205754

Change-Id: I3c2d6f0f2c06574c168c9473488b0474f96c8211
Signed-off-by: Anshuman Kar <anshumank@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/1705474
Reviewed-by: svc-mobile-coverity <svc-mobile-coverity@nvidia.com>
Reviewed-by: Shu Zhong <shuz@nvidia.com>
GVS: Gerrit_Virtual_Submit
Reviewed-by: Ujwal Patel <ujwalp@nvidia.com>
Reviewed-by: mobile promotions <svcmobile_promotions@nvidia.com>
Tested-by: mobile promotions <svcmobile_promotions@nvidia.com>
---

diff --git a/drivers/extcon/extcon.c b/drivers/extcon/extcon.c
index c97e2f5..858a683 100644
--- a/drivers/extcon/extcon.c
+++ b/drivers/extcon/extcon.c
@@ -14,6 +14,8 @@
  * Copyright (C) 2008 Google, Inc.
  * Author: Mike Lockwood <lockwood@android.com>
  *
+ * Copyright (c) 2018, NVIDIA CORPORATION, All rights reserved.
+ *
  * This software is licensed under the terms of the GNU General Public
  * License version 2, as published by the Free Software Foundation, and
  * may be copied, distributed, and modified under those terms.
@@ -173,6 +175,26 @@ struct __extcon_info {
 		.id = EXTCON_DISP_HMD,
 		.name = "HMD",
 	},
+	[EXTCON_DISP_AUDIO_AUX0] = {
+		.type = EXTCON_TYPE_DISP,
+		.id = EXTCON_DISP_AUDIO_AUX0,
+		.name = "AUDIO_AUX0",
+	},
+	[EXTCON_DISP_AUDIO_AUX1] = {
+		.type = EXTCON_TYPE_DISP,
+		.id = EXTCON_DISP_AUDIO_AUX1,
+		.name = "AUDIO_AUX1",
+	},
+	[EXTCON_DISP_AUDIO_AUX2] = {
+		.type = EXTCON_TYPE_DISP,
+		.id = EXTCON_DISP_AUDIO_AUX2,
+		.name = "AUDIO_AUX2",
+	},
+	[EXTCON_DISP_AUDIO_AUX3] = {
+		.type = EXTCON_TYPE_DISP,
+		.id = EXTCON_DISP_AUDIO_AUX3,
+		.name = "AUDIO_AUX3",
+	},
 
 	/* Miscellaneous external connector */
 	[EXTCON_DOCK] = {
diff --git a/include/linux/extcon.h b/include/linux/extcon.h
index 54804a3..3954481 100644
--- a/include/linux/extcon.h
+++ b/include/linux/extcon.h
@@ -74,6 +74,10 @@
 #define EXTCON_DISP_VGA		43	/* Video Graphics Array */
 #define EXTCON_DISP_DP		44	/* Display Port */
 #define EXTCON_DISP_HMD		45	/* Head-Mounted Display */
+#define EXTCON_DISP_AUDIO_AUX0	46	/* Audio for SOR 0 */
+#define EXTCON_DISP_AUDIO_AUX1	47	/* Audio for SOR 1 */
+#define EXTCON_DISP_AUDIO_AUX2	48	/* Audio for SOR 2 */
+#define EXTCON_DISP_AUDIO_AUX3	49	/* Audio for SOR 3 */
 
 /* Miscellaneous external connector */
 #define EXTCON_DOCK		60
diff --git a/include/linux/extcon/extcon-disp.h b/include/linux/extcon/extcon-disp.h
new file mode 100644
index 0000000..b4900f6
--- /dev/null
+++ b/include/linux/extcon/extcon-disp.h
@@ -0,0 +1,53 @@
+
+/*
+ * include/linux/extcon/extcon-disp.h
+ *
+ * extcon driver for display accessory detection compatible with switch-mid
+ *
+ * Copyright (c) 2018, NVIDIA CORPORATION, All rights reserved.
+ *
+ * This software is licensed under the terms of the GNU General Public
+ * License version 2, as published by the Free Software Foundation, and
+ * may be copied, distributed, and modified under those terms.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ */
+
+#ifndef _EXTCON_DISP_H_
+#define _EXTCON_DISP_H_ __FILE__
+
+#include <linux/module.h>
+#include <linux/extcon.h>
+
+#define EXTCON_DISP_HPD_STATE_DISABLED	false
+#define EXTCON_DISP_HPD_STATE_ENABLED	true
+#define EXTCON_DISP_AUX_STATE_DISABLED	false
+#define EXTCON_DISP_AUX_STATE_ENABLED	true
+
+#ifdef CONFIG_EXTCON_DISP_STATE
+/**
+ * disp_state_extcon_switch_report() - Report the state of external
+ * connector to extcon driver.
+ * @cable:	Cable for which state is reported.
+ * @state:	Reported state of cable
+ */
+void disp_state_extcon_switch_report(const unsigned int cable, bool state);
+/**
+ * disp_state_extcon_aux_report() - Wrapper on extcon switch report function
+ * to facilitate reporting aux connectors by their instance id.
+ * @aux_idx:	Audio connector instance for which state is reported
+ * @state:		Reported state of cable
+ */
+void disp_state_extcon_aux_report(const unsigned int aux_idx, bool state);
+#else
+static inline void disp_state_extcon_switch_report(const unsigned int cable,
+	bool state) { return; }
+static inline void disp_state_extcon_aux_report(const unsigned int aux_idx,
+	bool state) { return; }
+#endif
+
+#endif /* _EXTCON_DISP_H */
